packages/core/lib/console.js

```diff
@@ -16,18 +16,25 @@ const Require = require("@truffle/require");
 const debug = require("debug")("console");
 const { getCommand } = require("./command-utils");
 
-const processInput = input => {
-  const inputComponents = input.trim().split(" ");
-  if (inputComponents.length === 0) return input;
+const processInput = (input, allowedCommands) => {
+  const words = input.trim().split(/\s+/);
+  if (words.length === 0) return input;
 
-  if (inputComponents[0] === "truffle") {
-    return inputComponents.slice(1).join(" ");
+  if (words[0] === "truffle") {
+    if (allowedCommands.includes(words[1])) {
```

Comment:
Ah, there is another problem with this method of checking. If you implement it this way, it will display this message every time a user enters something that is not a command after "truffle". So `truffle tyler` should trigger this message. We probably want to just check for other Truffle commands that are not "allowed", right?

Reply:
Oh yea, you're correct!

Reply:
> If you type `truffle console`, apparently that also returns the `Console` object

Thanks, @gnidan. I updated the behavior so
- `truffle <restricted-command>` will log the command isn't allowed in the REPL.
- `truffle <unknown-command>` will log the command isn't a valid Truffle command.
- `any expression that doesn't start with truffle` will be interpreted by the REPL